process/common.go

```diff
@@ -700,54 +694,107 @@ func GetSortedStorageUpdates(account *vmcommon.OutputAccount) []*vmcommon.Storag
 	return storageUpdates
 }
 
-// GetScheduledSCRsFromStorage gets the scheduled scrs of the given header from storage
-func GetScheduledSCRsFromStorage(
+// GetScheduledRootHashAndSCRsFromStorage gets scheduled root hash and SCRs of the given header from storage
+func GetScheduledRootHashAndSCRsFromStorage(
 	headerHash []byte,
 	storageService dataRetriever.StorageService,
 	marshalizer marshal.Marshalizer,
-) (map[block.Type][]data.TransactionHandler, error) {
+) ([]byte, map[block.Type][]data.TransactionHandler, error) {
 	if check.IfNil(storageService) {
-		return nil, ErrNilStorage
+		return nil, nil, ErrNilStorage
 	}
 	if check.IfNil(marshalizer) {
-		return nil, ErrNilMarshalizer
+		return nil, nil, ErrNilMarshalizer
 	}
 
 	scheduledSCRsStorer := storageService.GetStorer(dataRetriever.ScheduledSCRsUnit)
 	if check.IfNil(scheduledSCRsStorer) {
-		return nil, ErrNilStorage
+		return nil, nil, ErrNilStorage
 	}
 
 	marshalizedSCRsBatch, err := scheduledSCRsStorer.Get(headerHash)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	b := &batch.Batch{}
 	err = marshalizer.Unmarshal(b, marshalizedSCRsBatch)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	scheduledRootHash := make([]byte, 0)
+	if len(b.Data) > 0 {
+		scheduledRootHash = b.Data[0]
 	}
 
 	mapScheduledSCRs := make(map[block.Type][]data.TransactionHandler)
-	for _, marshalizedScheduledSCRs := range b.Data {
-		scheduledSCRs := &ScheduledSCRs{}
+	for index, marshalizedScheduledSCRs := range b.Data {
+		if index == 0 {
```

Comment:
Index 0 represents the scheduled roothash?

Reply:
Yes